devito/operator/operator.py

```diff
@@ -871,11 +872,14 @@ def apply(self, **kwargs):
         # Post-process runtime arguments
         self._postprocess_arguments(args, **kwargs)
 
-        # Output summary of performance achieved
-        return self._emit_apply_profiling(args)
+        # In case MPI is used restrict result logging to one rank only
+        if configuration['mpi']:
```

Comment:
this shouldn't be needed

Reply:
This is needed only if the user accidentally asks for copies of an execution.
i.e. DEVITO_MPI=0 mpirun -n 2.
if not it will break with:

```bash
Attempting to use an MPI routine (internal_Comm_rank) before initializing or after finalizing MPICH
```

Reply:
I can get away with a try-except in mpi_switch_log ?

Reply:
probably

Reply:
added another condition in set_log_level